### PR TITLE
[#6569] Fixed issues with configuration reload tests (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Running core tests ...

### CODE REVIEW

File: lib/core/src/irods_server_properties.cpp [hunk 1]
```diff
@@ -257,24 +261,42 @@ namespace irods
...
+            if (std::any_of(std::begin(runtime_properties), std::end(runtime_properties), pred)) {
+                op = "add";
+                entry["value"] = config_props_.at(json::json_pointer{path});
             }
         }
-        auto lock = acquire_write_lock();
-        // TODO uncomment when issue #6470 is fixed
-        // log_server::error("Before patch '{}'", config_props_.dump());
+
+        log_server::debug("Before patch ==> {}", [this] { return std::make_tuple(config_props_.dump()); });
```
Comment (korydraughn) [inline lib/core/src/irods_server_properties.cpp:296]: Verify that this is correct and doesn't invoke UB.

File: server/main_server/src/rodsServer.cpp [hunk 2]
```diff
@@ -139,21 +139,21 @@ namespace
         // Should only ever set the cache salt once.
         try {
             const auto& existing_salt = irods::get_server_property<const std::string>(irods::KW_CFG_RE_CACHE_SALT);
-            log_server::error("createAndSetRECacheSalt: salt already set [{}]", existing_salt.c_str());
-            return ERROR(SYS_ALREADY_INITIALIZED, "createAndSetRECacheSalt: cache salt already set");
+            log_server::debug("createAndSetRECacheSalt: Cache salt already set [{}]", existing_salt.c_str());
```
Comment (alanking) [inline server/main_server/src/rodsServer.cpp:142]: Is this changing to `debug` because the returned `irods::error` will get logged somewhere?

Comment (trel) [reply server/main_server/src/rodsServer.cpp:142]: `debug` because this will be the normal case on a reload.

Comment (alanking) [reply server/main_server/src/rodsServer.cpp:142]: Okay

File: lib/core/src/irods_server_properties.cpp (hunk 1 above)
Comment (korydraughn) [reply lib/core/src/irods_server_properties.cpp:294]: Verified that the code works as intended.

Learned that the main server process does not honor the log level following a reload. Created issue at #6608.

**Review by alanking — APPROVED** [review]

**Comment by korydraughn** [comment]
All unit/core tests passed.

**Comment by korydraughn** [comment]
Added #'s

**Pull request merged**